lib/rules/no-unmodified-loop-condition.js

```diff
@@ -261,10 +261,10 @@ module.exports = function(context) {
 
             for (var j = 0; j < condition.groups.length; ++j) {
                 var be = condition.groups[j];
-                var group = groupMap.get(be);
+                var group = groupMap[be];
```

Comment:
Indeed the tests are lacking, though, I don't think we need replace the `Map` object by `lodash`. Because the `Map` object is a native object of ECMAScript. It's supported on Node since v0.12. We can just remove `es6-map` when we drop v0.10 support. And `escope` is using `es6-map`, so we cannot remove `es6-map` from dependencies even if removed from here.